src/coreclr/hosts/corebundle/TPA.cpp

```diff
@@ -0,0 +1,105 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <stdio.h>
+#include <windows.h>
+#include "TPA.h"
+#include "palclr.h"
+
+bool TPA::HasFile(_In_z_ char* fileNameWithoutExtension)
+{
+    if (!m_tpa.CStr())
+    {
+        return false;
+    }
+
+    for (int iExtension = 0; iExtension < countExtensions; iExtension++)
+    {
+        char fileName[MAX_LONGPATH];
+        strcpy_s(fileName, MAX_LONGPATH, "\\"); // So that we don't match other files that end with the current file name
+        strcpy_s(fileName, MAX_LONGPATH, fileNameWithoutExtension);
```

Comment:
@swaroop-sridhar it seems this should be `strcat_s` instead of `strcpy_s`, right?

Reply:
Correct, that's an oversight; I'll make a note to fix it when the code is ported to the product branch.

Reply:
Thanks for pointing it out.